fastnetmon fails to build with libndpi-dev 2.2-1

fastnetmon fails to build from source with libndpi-dev 2.2-1 on Debian unstable:
```
...
/usr/bin/c++  -DENABLE_DPI -DENABLE_LUA_HOOKS -DFASTNETMON_ENABLE_AFPACKET -DMONGO -DNETMAP_PLUGIN -DREDIS -I/usr/include/libndpi-2.2.0 -I/usr/include/luajit-2.1 -I/build/1st/fastnetmon-1.1.3+dfsg/src/netmap_plugin/netmap_includes -I/usr/include/libmongoc-1.0 -I/usr/include/libbson-1.0  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -o CMakeFiles/fastnetmon.dir/fastnetmon.cpp.o -c /build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp: In function 'void produce_dpi_dump_for_pcap_dump(std::__cxx11::string, std::stringstream&, std::__cxx11::string)':
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp:3635:85: error: 'ndpi_protocol {aka struct ndpi_proto}' has no member named 'protocol'; did you mean 'app_protocol'?
         char* protocol_name = ndpi_get_proto_name(my_ndpi_struct, detected_protocol.protocol);
                                                                                     ^~~~~~~~
                                                                                     app_protocol
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp:3638:31: error: 'ndpi_protocol {aka struct ndpi_proto}' has no member named 'protocol'; did you mean 'app_protocol'?
         if (detected_protocol.protocol == NDPI_PROTOCOL_DNS) {
                               ^~~~~~~~
                               app_protocol
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp:3644:38: error: 'ndpi_protocol {aka struct ndpi_proto}' has no member named 'protocol'; did you mean 'app_protocol'?
         } else if (detected_protocol.protocol == NDPI_PROTOCOL_NTP) {
                                      ^~~~~~~~
                                      app_protocol
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp:3649:38: error: 'ndpi_protocol {aka struct ndpi_proto}' has no member named 'protocol'; did you mean 'app_protocol'?
         } else if (detected_protocol.protocol == NDPI_PROTOCOL_SSDP) {
                                      ^~~~~~~~
                                      app_protocol
/build/1st/fastnetmon-1.1.3+dfsg/src/fastnetmon.cpp:3652:38: error: 'ndpi_protocol {aka struct ndpi_proto}' has no member named 'protocol'; did you mean 'app_protocol'?
         } else if (detected_protocol.protocol == NDPI_PROTOCOL_SNMP) {
                                      ^~~~~~~~
                                      app_protocol
CMakeFiles/fastnetmon.dir/build.make:65: recipe for target 'CMakeFiles/fastnetmon.dir/fastnetmon.cpp.o' failed
make[3]: *** [CMakeFiles/fastnetmon.dir/fastnetmon.cpp.o] Error 1
```
This bug was reported to Debian: https://bugs.debian.org/887623

--- a/src/fastnetmon.cpp
+++ b/src/fastnetmon.cpp
@@ -3699,24 +3699,29 @@ void produce_dpi_dump_for_pcap_dump(std::string pcap_file_path, std::stringstrea
 
         ndpi_protocol detected_protocol = dpi_parse_packet(packet_buffer, pcap_packet_header.orig_len, pcap_packet_header.incl_len, src, dst, flow, parsed_packet_as_string);
 
-        char* protocol_name = ndpi_get_proto_name(my_ndpi_struct, detected_protocol.protocol);
+#if NDPI_MAJOR >= 2
+        u_int16_t app_protocol = detected_protocol.app_protocol;
+#else
+        u_int16_t app_protocol = detected_protocol.protocol;
+#endif
+        char* protocol_name = ndpi_get_proto_name(my_ndpi_struct, app_protocol);
         char* master_protocol_name = ndpi_get_proto_name(my_ndpi_struct, detected_protocol.master_protocol); 
 
-        if (detected_protocol.protocol == NDPI_PROTOCOL_DNS) {
+        if (app_protocol == NDPI_PROTOCOL_DNS) {
             // It's answer for ANY request with so much
             if (flow->protos.dns.query_type == 255 && flow->protos.dns.num_queries < flow->protos.dns.num_answers) {
                 dns_amplification_packets++;
             }
 
-        } else if (detected_protocol.protocol == NDPI_PROTOCOL_NTP) {
+        } else if (app_protocol == NDPI_PROTOCOL_NTP) {
             // Detect packets with type MON_GETLIST_1
             if (flow->protos.ntp.version == 2 && flow->protos.ntp.request_code == 42) {
                 ntp_amplification_packets++;
             }
-        } else if (detected_protocol.protocol == NDPI_PROTOCOL_SSDP) {
+        } else if (app_protocol == NDPI_PROTOCOL_SSDP) {
             // So, this protocol completely unexpected in WAN networks
             ssdp_amplification_packets++;
-        } else if (detected_protocol.protocol == NDPI_PROTOCOL_SNMP) {
+        } else if (app_protocol == NDPI_PROTOCOL_SNMP) {
             // TODO: we need detailed tests for SNMP!
             snmp_amplification_packets++;
         }

